server/server.go

```diff
@@ -280,7 +285,13 @@ func (s *Server) restore(ctx context.Context) []storage.StorageImageID {
 		}
 	}
 
-	// Cleanup the deletedPods in the networking plugin
+	// First cleanup any potential stale network resources not associated to any
+	// pod known to us
+	if err := s.networkGC(ctx, knownPods); err != nil {
```

Comment:
so unfortunately this will run before the networking plugin has a chance to startup, which is why the other pieces are wrapped in the resource cleaner below. 

can you adapt https://github.com/cri-o/cri-o/pull/6547 into here, and call the GC first, and then loop through the pods all in one routine?

Reply:
Thanks, wasn't aware of that.

The extra complication is that we can't be attending new pod requests as the list of valid pods needs to be stable while we call GC. I will see how we can go about this.

Reply:
an option is you can create all of the pods that are hostNetwork, then run GC, then create the rest?

Reply:
@haircommander 

I went with a (maybe naive) option of calling GC in the background while holding a lock that prevents running non host network pods in the meantime.

I didn't adopt your code as there was no good reason for the specific order, both GC and DEL should be able to run parallel.

Reply:
I could adopt your could though if that still helps.